cli/pcluster/config/param_types.py

```diff
@@ -1224,16 +412,28 @@ def from_file(self, config_parser, fail_on_absence=False):
         params_definitions = self.definition.get("params")
         section_name = get_file_section_name(self.key, self.label)
 
+        # Only params with PUBLIC visibility can be specified in config file
+        public_param_keys = [
```

Comment:
minor: if we use a set rather than a list here the lookup below is faster

Reply:
Done